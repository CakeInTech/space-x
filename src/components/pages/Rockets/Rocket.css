.container {
    border: 2px solid red;
}

.eachbox {
    border: 2px solid black;
}