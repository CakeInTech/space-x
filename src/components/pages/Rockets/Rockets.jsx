import React from 'react';
import { useDispatch } from 'react-redux';
import { fetchRockets } from '../../../redux/rockets/rockets';

const Rockets = () => {
  const dispatch = useDispatch();
  return (
  
  <div>
    <h1>This is the rockets page</h1>
    <button onClick={() => dispatch(fetchRockets())}>Click to log out data</button>
  </div>
);
}

export default Rockets;
