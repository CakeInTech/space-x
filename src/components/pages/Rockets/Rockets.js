import React from 'react';
import { useDispatch } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';
import {
  Container, Row, Col, Image, Button, Collapse,
} from 'react-bootstrap';
import PropTypes from 'prop-types';
import { buttonToggle } from '../../../redux/rockets/rockets';
import './Rocket.css';

const Rockets = (props) => {
  const dispatch = useDispatch();
  const { rockets } = props;
  return (
    <Container fluid className="container">
      {rockets.map((element) => (
        <Row key={uuidv4()}>
          <Col className="eachbox" xs={5}><Image fluid src={element.flickr_images} alt="Flicker Image" /></Col>
          <Col className="eachbox">
            <h1>{element.name}</h1>
            <Collapse in={element.reserved}><Button variant="info" type="button" size="sm">Reserved</Button></Collapse>
            <p>{element.description}</p>
            {
              (element.reserved)
                ? <Button id={element.id} variant="light" className="cancelButton" type="button" onClick={(e) => dispatch(buttonToggle(e.target.id))}>Cancel Reservation</Button>
                : <Button id={element.id} variant="primary" type="button" onClick={(e) => dispatch(buttonToggle(e.target.id))}>Reserve Rocket</Button>
            }
          </Col>
        </Row>
      ))}
    </Container>
  );
};

Rockets.propTypes = {
  rockets: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    flickr_images: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reserved: PropTypes.bool.isRequired,
  })).isRequired,
};

export default Rockets;
