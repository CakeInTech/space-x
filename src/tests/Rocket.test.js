import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rockets from '../components/pages/Rockets/Rockets';

it('Test Mission component', () => {
  const Rocketeer = render(
    <Provider store={store}>
      <Router>
        <Rockets />
      </Router>
    </Provider>,
  );
  expect(Rocketeer).toMatchSnapshot();
});
