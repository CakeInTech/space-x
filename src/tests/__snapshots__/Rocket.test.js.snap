// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Mission component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="eachbox col-5"
          >
            <img
              alt="Flicker Image"
              class="img-fluid"
              src=" "
            />
          </div>
          <div
            class="eachbox col"
          >
            <h1>
               
            </h1>
            <button
              class="collapse btn btn-info btn-sm"
              type="button"
            >
              Reserved
            </button>
            <p>
               
            </p>
            <button
              class="btn btn-primary"
              id=" "
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="eachbox col-5"
        >
          <img
            alt="Flicker Image"
            class="img-fluid"
            src=" "
          />
        </div>
        <div
          class="eachbox col"
        >
          <h1>
             
          </h1>
          <button
            class="collapse btn btn-info btn-sm"
            type="button"
          >
            Reserved
          </button>
          <p>
             
          </p>
          <button
            class="btn btn-primary"
            id=" "
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
